src/main/scala/com/cloudera/sparkts/UnivariateTimeSeries.scala

```diff
@@ -326,5 +326,108 @@ object UnivariateTimeSeries {
     }
     sampledValues
   }
+
+  /**
+   * Difference a vector with respect to the m-th prior element. Size-preserving by leaving first
+   * `m` elements intact. This is the inverse of the `inverseDifferences` function.
+   * @param ts Series to difference
+   * @param lag The difference lag (e.g. x means y(i) = ts(i) - ts(i - x), etc)
+   * @return a new differenced vector
+   */
+  def differencesAtLag(ts: Vector[Double], lag: Int): Vector[Double] = {
+    if (lag == 0) {
+      // for consistency, since we create a new vector in else-branch
+      ts.copy
+    } else {
+      val n = ts.length
+      val diffedTs = new DenseVector(Array.fill(n)(0.0))
+      var i = 0
+
+      while (i < n) {
+        // elements prior to `lag` are copied over without modification
+        diffedTs(i) = if (i < lag) ts(i) else ts(i) - ts(i - lag)
+        i += 1
+      }
+      diffedTs
+    }
+  }
+
+  /**
+   * Calculate an "inverse-differenced" vector of a given lag. Size-preserving by leaving first
+   * `lag` elements intact. This is the inverse of the `differences` function.
+   * @param ts Series to add up
+   * @param lag The difference lag to add (e.g. x means y(i) = ts(i) + y(i -
+   *              x), etc)
+   * @return a new vector where the difference operation as been inverted
+   */
+  def inverseDifferencesAtLag(ts: Vector[Double], lag: Int): Vector[Double] = {
+    if (lag == 0) {
+      // for consistency, since we create a new vector in else-branch
+      ts.copy
+    } else {
+      val n = ts.length
+      val addedTs = new DenseVector(Array.fill(n)(0.0))
+      var i = 0
+
+      while (i < n) {
+        // elements prior to `order` are copied over without modification
+        addedTs(i) = if (i < lag) ts(i) else ts(i) + addedTs(i - lag)
+        i += 1
+      }
+      addedTs
+    }
+  }
+
+  /**
+   * Performs differencing of order `d`. This means we recursively difference a vector a total of
+   * d-times. So that d = 2 is a vector of the differences of differences. Note that for each
+   * difference level, d_i, the element at ts(d_i - 1) corresponds to the value in the prior
+   * iteration.
+   * @param ts time series to difference
+   * @param d order of differencing
+   * @return a vector of the same length differenced to order d
+   */
+  def differencesOfOrderD(ts: Vector[Double], d: Int): Vector[Double] = {
```

Comment:
Also just realized I should just be creating a copy of `ts` prior to calling the helper, rather than in the helper function (which means the vector gets copied with every recursive call). My bad. I will fix this once we decide on the better approach overall.